patches/tModLoader/Terraria/Program.TML.cs

```diff
@@ -120,6 +120,44 @@ private static void SetSavePath()
 			Logging.tML.Info($"Controlled Folder Access feature detected. If game fails to launch make sure to add \"{Environment.ProcessPath}\" to the \"Allow an app through Controlled folder access\" menu found in the \"Ransomware protection\" menu."); // Before language is loaded, no need to localize
 	}
 
+	private static void StartupSequenceTml(bool isServer)
+	{
+		try {
+			ControlledFolderAccessSupport.CheckFileSystemAccess();
+			Logging.Init(isServer ? Logging.LogFile.Server : Logging.LogFile.Client);
+
+			if (Platform.Current.Type == PlatformType.Windows && System.Runtime.InteropServices.RuntimeInformation.ProcessArchitecture != System.Runtime.InteropServices.Architecture.X64)
+				throw new Exception("The current Windows Architecture of your System is CURRENTLY unsupported. Aborting...");
+		}
+		catch (Exception e) {
+			ErrorReporting.FatalExit("Failed to init logging", e);
+		}
+
+		Logging.LogStartup(isServer); // Should run as early as is possible. Want as complete a log file as possible
+
+		try {
+			SetSavePath(); 
+		}
+		catch (Exception e) {
+			ErrorReporting.FatalExit("Failed to establish a save location", e);
+		}
+				
+		if (ModLoader.Core.ModCompile.DeveloperMode) // Needs to run after SetSavePath
+			Logging.tML.Info("Developer mode enabled");
+
+		AttemptSupportHighDPI(isServer); // Can run anytime
+
+		try {
+			CheckDependencies(); // Should run after LogStartup
+		}
+		catch (Exception e) {
+			ErrorReporting.FatalExit("Unexpected failure in verifying dependencies. Please reach out in the tModLoader Discord for support", e);
```

Comment:
No need for `try catch` here, should never be hit, the handlers are all in `CheckNativeFAudioDependencies`

Reply:
If you still want a fallback handler, a blanket `try-catch` around the entire `StartupSequenceTml` method would be more appropriate